server/src/test/java/org/elasticsearch/action/search/CanMatchPreFilterSearchPhaseTests.java

```diff
@@ -356,7 +356,10 @@ public void run() {
                 .sorted(Comparator.comparing(minAndMaxes::get, MinAndMax.getComparator(order)).thenComparing(shardIds::get))
                 .map(shardIds::get)
                 .toArray(ShardId[]::new);
-
+            if (shardToSkip.size() == expected.length) {
+                // we need at least one shard to produce the empty result for aggs
```

Comment:
For my own learning: this is the special case mentioned in CanMatchPreFilterSearchPhase#getIterator where we add at least one shard index to the set to be queried? And that is the reason why we need to remove that one from the skipped set here in the test?

Reply:
yep, you got it right.